lib/stdlib/uc_spec/gen_unicode_mod.escript

```diff
@@ -0,0 +1,834 @@
+            #cp{cs={U,L,T,_}} = Entry = array:get(CP, Table),
+            array:set(CP, Entry#cp{cs={U,L,T,Case}}, Table)
+    end.
+
+unlist([A]) -> A;
+unlist(A) -> A.
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+parse_comp_excl(Line, Table) ->
+    [CodePoint|_Comments] = tokens(Line, "#"),
+    CP = hex_to_int(CodePoint),
+    Entry = array:get(CP, Table),
+    array:set(CP, Entry#cp{dec=none}, Table).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+parse_properties(Line0, Acc) ->
+    [Line|_Comments] = tokens(Line0, "#"),
+    [CodePoints, Class] = tokens(Line, ";"),
+    case tokens(CodePoints, ".") of
+        [CodePoint] ->
+            [{to_atom(Class), {hex_to_int(CodePoint), undefined}}|Acc];
+        [CodePoint1,"",CodePoint2] ->
+            [{to_atom(Class), {hex_to_int(CodePoint1), hex_to_int(CodePoint2)}}|Acc]
+    end.
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+gen_file(Fd, Data, ExclData, Props) ->
+    gen_header(Fd),
+    gen_static(Fd),
+    gen_norm(Fd),
+    gen_ws(Fd, Props),
+    gen_cp(Fd),
+    gen_gc(Fd, Props),
+    gen_compose_pairs(Fd, ExclData, Data),
+    gen_case_table(Fd, Data),
+    gen_unicode_table(Fd, Data),
+    ok.
+
+gen_header(Fd) ->
+    io:put_chars(Fd, "%%\n%% this file is generated do not modify\n"),
+    io:put_chars(Fd, "%% see ../uc_spec/gen_unicode_mod.escript\n\n"),
+    io:put_chars(Fd, "-module(" ++ ?MOD ++").\n"),
+    io:put_chars(Fd, "-export([cp/1, gc/1]).\n"),
+    io:put_chars(Fd, "-export([nfd/1, nfc/1, nfkd/1, nfkc/1]).\n"),
+    io:put_chars(Fd, "-export([whitespace/0, is_whitespace/1]).\n"),
+    io:put_chars(Fd, "-export([uppercase/1, lowercase/1, titlecase/1, casefold/1]).\n\n"),
+    io:put_chars(Fd, "-export([spec_version/0, lookup/1, get_case/1]).\n"),
+    io:put_chars(Fd, "-inline([class/1]).\n"),
+    io:put_chars(Fd, "-compile(nowarn_unused_vars).\n"),
+    io:put_chars(Fd, "-dialyzer({no_improper_lists, cp/1}).\n"),
+    io:put_chars(Fd, "-type gc() :: char()|[char()].\n\n\n"),
+    ok.
+
+gen_static(Fd) ->
+    io:put_chars(Fd, "-spec lookup(char()) -> #{'canon':=[{byte(),char()}], 'ccc':=byte(), "
+                 "'compat':=[] | {atom(),[{byte(),char()}]}}.\n"),
+    io:put_chars(Fd, "lookup(Codepoint) ->\n"
+                 "    {CCC,Can,Comp} = unicode_table(Codepoint),\n"
+                 "    #{ccc=>CCC, canon=>Can, compat=>Comp}.\n\n"),
+    io:put_chars(Fd, "-spec get_case(char()) -> #{'fold':=gc(), 'lower':=gc(), 'title':=gc(), 'upper':=gc()}.\n"),
+    io:put_chars(Fd, "get_case(Codepoint) ->\n"
+                 "    case case_table(Codepoint) of\n"
+                 "        {U,L} -> #{upper=>U,lower=>L,title=>U,fold=>L};\n"
+                 "        {U,L,T,F} -> #{upper=>U,lower=>L,title=>T,fold=>F}\n"
+                 "    end.\n\n"),
+    io:put_chars(Fd, "spec_version() -> {9,0}.\n\n\n"),
```

Comment:
Maybe move this to a `-define` so it is more straight-forward to update when the unicode files are updated?